dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/passwordreset/PasswordResetServiceBean.java

```diff
@@ -73,6 +76,9 @@ public PasswordResetInitResponse requestPasswordReset( BuiltinUser aUser, boolea
         
         // create a fresh token for the user
         PasswordResetData passwordResetData = new PasswordResetData(aUser);
+        // set token expiration date based on SystemSettings
+        updateTokenExpirationTime(aUser);
```

Comment:
This method won't work, info below.

Reply:
You can just do passwordResetData.setExpires(TimeUnit.MINUTES.toMillis(
systemConfig.getMinutesUntilPasswordResetTokenExpires())).